pkg/metrics/metrics.go

```diff
@@ -94,11 +98,11 @@ var (
 		[]string{"outcome"})
 
 	// EndpointRegenerationTime is the total time taken to regenerate endpoint
-	EndpointRegenerationTime = prometheus.NewCounter(prometheus.CounterOpts{
+	EndpointRegenerationTime = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: Namespace,
-		Name:      "endpoint_regeneration_seconds_total",
```

Comment:
We can't just rename metrics as it will break dashboards. We need to add a new one and declare the old one to be obsoleted in 6 months or similar